Remove `extra-substituters` and `extra-trusted-public-keys` from nix (#670)
Plutonomicon/plutarch-plutus: plutarch-docs/README.md

- [Common Extensions and GHC options](./Run.md#common-extensions-and-ghc-options)
- [Evaluation](./Run.md#evaluation)
- Nix binary cache is recommended. Otherwise nix will build from GHC itself.

Use Plutonomicon binary cache via cachix: https://plutonomicon.cachix.org

> When setting up cachix, it recommends adding users to `trusted-users` at nix configuration. This
> is [insecure](https://github.com/cachix/cachix/issues/612) and is not recommended.

Reviewer: ```suggestion
DO NOT add your user to `trusted-users` as `cachix use` suggests because this is
[insecure](https://nixos.org/manual/nix/stable/command-ref/conf-file.html#conf-trusted-users).
```
